geopandas/array.py

```diff
@@ -828,15 +828,34 @@ def estimate_utm_crs(self, datum_name="WGS 84"):
             raise RuntimeError("crs must be set to estimate UTM CRS.")
 
         minx, miny, maxx, maxy = self.total_bounds
+        x_center = None
+        y_center = None
         # ensure using geographic coordinates
         if not self.crs.is_geographic:
-            lon, lat = Transformer.from_crs(
-                self.crs, "EPSG:4326", always_xy=True
-            ).transform((minx, maxx, minx, maxx), (miny, miny, maxy, maxy))
-            x_center = np.mean(lon)
-            y_center = np.mean(lat)
-        else:
+            transformer = Transformer.from_crs(self.crs, "EPSG:4326", always_xy=True)
+            if compat.PYPROJ_GE_31:
+                minx, miny, maxx, maxy = transformer.transform_bounds(
+                    minx, miny, maxx, maxy
+                )
+                # crossed the antimeridian
+                if minx > maxx:
+                    # shift maxx from [-180,180] to [0,360]
+                    # so both numbers are positive for center calculation
+                    # Example: -175 to 185
+                    maxx += 360
+                    x_center = np.mean([minx, maxx])
+                    # shift back to [-180,180]
+                    x_center = ((x_center + 180) % 360) - 180
```

Comment:
```suggestion
                    x_center = ((x_center + 180) % 360) - 180
                else:
                    x_center = np.mean([minx, maxx])
                y_center = np.mean([miny, maxy])
```

(then you don't need to check if it's None below (which makes it IMO a bit harder to follow), but use a plain `else` instead)

Reply:
Sounds good to me :+1: